src/main/java/io/fabric8/elasticsearch/util/RequestUtils.java

```diff
@@ -73,6 +81,42 @@ public boolean isOperationsUser(RestRequest request) {
         }
         return allowed;
     }
+    
+    @SuppressWarnings("rawtypes")
+    public void assertUser(RestRequest request) throws Exception {
+        final String user = getUser(request);
+        final String token = getBearerToken(request);
+        ConfigBuilder builder = new ConfigBuilder().withOauthToken(token);
+        try (DefaultOpenShiftClient osClient = new DefaultOpenShiftClient(builder.build())) {
+            LOGGER.debug("Verifying user {} matches the given token.", user);
+            Request okRequest = new Request.Builder()
+                .addHeader(AUTHORIZATION_HEADER, "Bearer " + token)
+                .url(osClient.getMasterUrl() + "oapi/v1/users/~") 
+                .build();
+            String username = null;
+            try (Response response = osClient.getHttpClient().newCall(okRequest).execute()){
+                if(response.code() != RestStatus.OK.getStatus()) {
+                    throw new ElasticsearchSecurityException("The user is unauthorized", RestStatus.UNAUTHORIZED);
+                }
+                String body = response.body().string();
+                if (LOGGER.isDebugEnabled()) {
+                    LOGGER.debug("Response: {}", new String(body));
```

Comment:
Isn't `body` already a `String`?

Reply:
Not its a responsebody type or something specific to the library impl

Reply:
But on line 101 `body` appears to be defined as a `String`.  So why do we need to do `new String(body)` now.